wrappers/core18.go

```diff
@@ -485,6 +490,53 @@ func undoSnapdUserServicesOnCore(s *snap.Info, inter interacter) error {
 	return nil
 }
 
+func writeSnapdDbusConfigOnCore(s *snap.Info) error {
+	// list D-Bus configuration fragments in the snapd snap
+	systemConfigs, err := filepath.Glob(filepath.Join(s.MountDir(), "usr/share/dbus-1/system.d/*.conf"))
+	if err != nil {
+		return err
+	}
+	sessionConfigs, err := filepath.Glob(filepath.Join(s.MountDir(), "usr/share/dbus-1/session.d/*.conf"))
+	if err != nil {
+		return err
+	}
+
+	state := make(map[string]osutil.FileState, len(systemConfigs)+1)
+	for _, config := range systemConfigs {
+		state[filepath.Base(config)] = &osutil.FileReference{
+			Path: config,
+		}
+	}
+	_, _, err = osutil.EnsureDirState(dirs.SnapDBusSystemPolicyDir, "snapd.*.conf", state)
+	if err != nil {
+		// TODO: uhhhh, what do we do in this case?
+		return err
+	}
+
+	state = make(map[string]osutil.FileState, len(sessionConfigs)+1)
+	for _, config := range sessionConfigs {
+		state[filepath.Base(config)] = &osutil.FileReference{
+			Path: config,
+		}
+	}
+	_, _, err = osutil.EnsureDirState(dirs.SnapDBusSessionPolicyDir, "snapd.*.conf", state)
+	if err != nil {
+		// TODO: uhhhh, what do we do in this case?
```

Comment:
Before this goes in I would love to at least see an updated TODO comment about the expected plan for this case.

Reply:
I cargo culted that TODO from one of the other routines in the file.  Thinking about it a bit more, I don't think there is anything more to do here.  All the changes made by this function will be undone by `RemoveSnapdSnapServicesOnCore`, which gets called while rolling back the install of the snapd snap.

We don't currently have any logic besides the `EnsureDirState` calls here, so I think the error handling is fine as is.